_posts/2019-03-14-2019-kafka-connector.md

```diff
@@ -0,0 +1,217 @@
+          - name: kafka_broker
+            value: "kafka"
+        volumeMounts:
+            - name: auth
+              readOnly: true
+              mountPath: "/var/secrets/"
+      volumes:
+      - name: auth
+        secret:
+          secretName: basic-auth
+```
+
+If you deployed your OpenFaaS instance with Helm and default values there is no need to make changes to this file. The `kafka_broker` variable is our broker host from which we will pull topics.
+
+In our case the broker is located within a service called `kafka`, which resides in the same namespace as the OpenFaaS gateway and uses the default kafka port of `9092`.
+
+The environmental variable `topics` is the place where we list the topics in which we are interested to receive from the broker. We will be using single topic for our example, but feel free to use as much as you wish, just separate them with comma like so `topics=example-topic-1,example-topic-2`.
+
+## Deploy the components
+
+In order to deploy the whole stack and play around with it go ahead and apply the kubernetes files while inside the kafka-connector folder:
+
+```bash
+$ kubectl apply -f yaml/kubernetes
+```
+The Broker deployed from the stack above contains a Producer in itself, but it is nested inside folders in the pods, so instead of searching it we will deploy simple producer provided by Alex:
+
+```bash
+$ kubectl run -n openfaas -t -i --image alexellis2/kafka-producer:0.2.0 producer
+```
+
+Congratulations now you have the whole stack deployed:
+* Zookeeper
+* Producer Service
+* Broker Host
+* Kafka Connector
+
+Before we proceed, make sure you have all the components running:
+
+```bash
+$ kubectl get pods -n openfaas
+NAME                               READY   STATUS    RESTARTS   AGE
+alertmanager-b55c578d4-87dgf       1/1     Running   0          37m
+faas-idler-7945c9bc7b-74z68        1/1     Running   1          37m
+gateway-7b7455dfd8-kzktf           2/2     Running   1          37m
+kafka-broker-544fbccd48-b5f65      1/1     Running   0          33m
+kafka-connector-7898548745-5wclq   1/1     Running   0          33m
+nats-6686bb4b95-lj6b8              1/1     Running   0          37m
+producer-c77fb7ccf-cx8xg           1/1     Running   0          31m
+prometheus-7dd5d65568-bkr8j        1/1     Running   0          37m
+queue-worker-8655bb54cb-q5nk8      1/1     Running   1          37m
+zookeeper-699f568f6f-wpzb6         1/1     Running   0          33m
+
+```
+
+## Subscribe function
+
+In order to consume topics via the connector we simply need to apply an annotation with a key of `topic` and the chosen value used in the `connector-dep.yml` - in our case it was `faas-request`.
+
+Here it is important to note that if you would like your function to subscribe to multiple topics, you just need to list them as we did in the connector configuration.
+
+We will use function from the store called `figlet`:
+
+```bash
+$ faas-cli store deploy figlet --annotation "topic=faas-request"
```

Comment:
I like this example, but I think we need an example with a YAML file too and a custom function.

Reply:
Done 👍